fdbclient/Versionstamp.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Versionstamp.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FDBCLIENT_VERSIONSTAMP_H
+#define FDBCLIENT_VERSIONSTAMP_H
+
+#pragma once
+
+#include "flow/Arena.h"
+
+const size_t VERSIONSTAMP_TUPLE_SIZE = 12;
+
+struct Versionstamp {
+	Versionstamp(StringRef);
+
+	int64_t getVersion() const;
+	int16_t getBatchNumber() const;
+	int16_t getClientWrittenNumber() const;
```

Comment:
I believe we call this the "user version" in some of the other bindings. I suppose `clientVersion` would also be fine, but `ClientWrittenNumber` is a little awkward